### PR TITLE
Fix retroactive conformances in tests

### PR DESCRIPTION
**Pull request by MahdiBM**
Currently results in warnings when building the tests.

### CODE REVIEW

File: Tests/MustacheTests/TemplateParserTests.swift
Comment (Joannis) [inline Tests/MustacheTests/TemplateParserTests.swift:66]: It would be nicer to put then`==` implementation in a singular place, and then add the conformance separately under compiler directive. Also, you can use `#if hasFeature(RetroactiveAttribute)`

File: Tests/MustacheTests/TemplateParserTests.swift
Comment (Joannis) [inline Tests/MustacheTests/TemplateParserTests.swift:95]: Same here

**Review by Joannis — CHANGES_REQUESTED** [review]
Thanks for the PR! Just one note

File: Tests/MustacheTests/TemplateParserTests.swift
Comment (MahdiBM) [reply Tests/MustacheTests/TemplateParserTests.swift]: I was aware of `#if hasFeature(RetroactiveAttribute)` as I originally suggested it to Adam in a PR, but In this PR, i was going based on https://github.com/apple/swift-nio/pull/2719 which mentions using `#if compiler()`.

However, after double-checking things again myself, I think `#if hasFeature(RetroactiveAttribute)` is still pretty much correct, and more clear than `#if compiler()`. So I'll change things to use that instead.

**Review by Joannis — APPROVED** [review]
I'm happier with this :) I'll wait for Adam's input before merge

**Review by adam-fowler — APPROVED** [review]

**Pull request merged**
